engine/tester/tgorm/gorm_db_for_printing_sql.go

```diff
@@ -0,0 +1,78 @@
+package tgorm
+
+import (
+	"context"
+	"database/sql"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+)
+
+// DBType Supported databases by this package.
+// Added here to prevent circular dependencies.
+type DBType string
+
+const (
+	// PostgreSQL is the PostgreSQL database type.
+	PostgreSQL DBType = "postgresql"
+	// SQLite is the SQLite database type.
+	SQLite DBType = "sqlite"
+)
+
+// GormDBForPrintingSQL is creating a gorm.DB instance that can be used to print or check SQL (without connecting to postgres for example).
+// This is useful for checking generated SQL statements.
+func GormDBForPrintingSQL(dbType DBType) *gorm.DB {
+	dialector := chooseDialector(dbType)
+
+	db, err := gorm.Open(dialector, gormConfig())
+	if err != nil {
+		panic(err)
+	}
+
+	return db
+}
+
+func chooseDialector(dbType DBType) gorm.Dialector {
+	switch dbType {
+	case PostgreSQL:
+		return postgres.New(postgres.Config{
+			Conn: &doNothingConnectionPool{},
+		})
+	case SQLite:
+		return sqlite.Open(":memory:")
+	default:
+		panic("unsupported database type " + dbType)
+	}
+}
+
+func gormConfig() *gorm.Config {
+	return &gorm.Config{
+		DryRun:         true,
+		TranslateError: true,
+		NamingStrategy: schema.NamingStrategy{
+			TablePrefix:   "xapi_",
+			SingularTable: false,
+		},
+	}
+}
+
+type doNothingConnectionPool struct {
+}
+
+func (m *doNothingConnectionPool) PrepareContext(_ context.Context, _ string) (*sql.Stmt, error) {
```

Comment:
What's the use case for those functions?

Reply:
As additional context, we use this in 2 places:
1. the tests for datastore custom `where clause`  - which is inheritance from BUX and we introduce those tests to ensure that actually generated SQL is SQLInjection proof (yes, because it wasn't previously)
2. I use it in examples for selector to show the generated sql query, because the query isn't trivial, and building it with gorm doesn't actually make it more readable, therefore thanks to testable examples, one can just copy the SQL from `// Output: ` and use it against the chosen database to check the result of the query, and eventually format the SQL string so that it would be more readable for him.